test/e2e/storage/volume_provisioning.go

```diff
@@ -351,6 +351,10 @@ var _ = utils.SIGDescribe("Dynamic Provisioning", func() {
 					continue
 				}
 
+				if zone, ok := test.Parameters["zone"]; ok {
+					framework.ExpectNotEqual(len(zone), 0, "expect at least one zone")
```

Comment:
Do you want the test to fail, or be skipped?

Reply:
Should fail.
I should explained this more clearly, let me clarify why this change is needed.

Originally, the testcase will fail with the local provider, this does not make sense since this testcase should be skipped instead, code snippet right above this line have that logic, 
```
framework.ProviderIs(test.CloudProviders...)   // local provider is not belonged to `test.CloudProviders` which should be the provider like `gce` as an example.
```

But the testcase failed instead of skipped because the `getRandomClusterZone` is called before the checking of the provider. 

`getRandomClusterZone` check the `zones` which generate a map based on the below node labels,
```
failure-domain.beta.kubernetes.io/zone
topology.kubernetes.io/zone
```

local provider doesn't have necessary have the labels set on the host, so the testcase failed.

What I did was extracting the line of the code for zones checking out and defer this check to the end of the provider validation.

The motivation behind this is we are trying to enable a CI system locally and run all those storage testcases with local provider as the first step, so we need to cleanup all failures of testcases which should not fail at all.

Reply:
when the testcase hit this line means the provider is not `local` and zone label is not set, so the testcase should fail, this is an expected behavior.

and if the provider is `local`, the testcase won't hit this line at all, it should be skipped due to the provider check.